src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> element = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = element;
+        } else {
+            tail.next = element;
+            element.prev = tail;
+            tail = element;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAddition(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> current = (index < size / 2) ? getNodeFromHead(index) : getNodeFromTail(index);
+        Node<T> newElement = new Node<>(current.prev, value, current);
+        if (current.prev != null) {
```

Comment:
Don't complicate if-else construction. Since you are returning from the `if` block, you don't need an `else` block, just continue with the code.